inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -219,34 +219,50 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
     bool workModeAuto = workMode != fullConfig.end() && workMode->second == InferenceEngine::PluginConfigParams::YES;
     auto priorities = fullConfig.find(MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES);
 
-    // not found device priorities for -d AUTO use case
-    if (priorities == fullConfig.end()) {
-        if (workModeAuto) {
-            std::string allDevices;
-            auto availableDevices = GetCore()->GetAvailableDevices();
-            if (availableDevices.empty()) {
-                IE_THROW(NotFound) << "No available device found";
-            }
-            for (auto&& device : availableDevices) {
-                allDevices += device;
-                allDevices += ((device == availableDevices[availableDevices.size()-1]) ? "" : ",");
-            }
-            metaDevices = ParseMetaDevices(allDevices, fullConfig);
-            multiNetworkConfig.insert({MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, allDevices});
-        } else {
-            IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for " << GetName() << " device";
+    // if workMode is AUTO
+    if (workModeAuto) {
+        // check the configure and check if need to set PerfCounters configure to device
+        // and set filter configure
+        bool needPerfCounters = false;
+        std::map<std::string, std::string> filterConfig;
+        CheckConfig(fullConfig, needPerfCounters, filterConfig);
+        // filter the device that supports filter configure
+        auto strDevices = GetDeviceList(fullConfig);
+        auto metaDevices = ParseMetaDevices(strDevices, fullConfig);
+        auto supportDevices = FilterDevice(metaDevices, filterConfig);
+        if (supportDevices.size() == 0) {
+             IE_THROW() << "there is no device support the configure";
+        }
+        // replace the configure with configure that auto want to pass to device
+        // and reset the strDevices to support devices
+        std::vector<std::string> validConfigKey;
```

Comment:
minor but better to do std::vector<std::string> validConfigKey =  PerfHintsConfig::SupportedKeys(); rather than explicitly add the specific values in the lines 241-242 below ( since if the new config is added to the hints impl, you will need to explicitly add that)